src/main/java/org/jenkinsci/plugins/reverse_proxy_auth/ReverseProxySecurityRealm.java

```diff
@@ -439,9 +440,7 @@ public static String toProviderUrl(@CheckForNull String serverUrl, @CheckForNull
 		return buf.toString();
 	}
 
-	public String getManagerPassword() {
-		return Scrambler.descramble(managerPassword);
```

Comment:
Removing `Scrambler#descramble` here looks like it will break all existing configurations.

Example:

Actual password: `thePassword`
Previously persisted password:  `dGhlUGFzc3dvcmQ=` ("scrambled" form of `thePassword`)

```
$ echo 'dGhlUGFzc3dvcmQ=' | base64 -D
thePassword
```

PR 66 persisted password after upgrade: encrypted form of `dGhlUGFzc3dvcmQ=` instead of encrypted form of `thePassword`

So having a previously configured LDAP password, this will change it from normal string to scrambled string for users.

Either have explicit migration (e.g. `readResolve`) or store the scrambled password encrypted.

I also recommend you create a `@JenkinsRule` based test loading an old `config.xml` (`@LocalData`) with password defined, and confirm it returns the expected plain text value from `getManagerPassword()`.

Reply:
Good point, Daniel. I will have a look.